librz/util/buf.c

```diff
@@ -1368,7 +1412,21 @@ RZ_API ut64 rz_buf_fwd_scan(RZ_NONNULL RzBuffer *b, ut64 start, ut64 amount, RZ_
 	if (!amount) {
 		return 0;
 	}
-	if (b->methods->get_whole_buf) {
+	if (b->type == RZ_BUFFER_BYTES || b->type == RZ_BUFFER_MMAP) {
+		ut64 sz;
+		const ut8 *buf;
```

Comment:
define these at the beginning of the function and remove the other definitions within the if/else.